test/unit/StateHoldrTests/setCollection.js

```diff
@@ -0,0 +1,41 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("sets collectionKeyRaw", function () {
+            // Arrange
+            var StateHolder = mocks.mockStateHoldr();
+
+            // Act
+            StateHolder.setCollection("newCollection", "value");
+
+            // Assert
+            expect(StateHolder.getCollectionKeyRaw()).to.equal("newCollection");
+        });
+
+        it("sets collectionKey", function () {
+            // Arrange
+            var StateHolder = mocks.mockStateHoldr({
+                prefix: "prefix",
+                ItemsHolder: mocks.mockItemsHoldr()
+            });
+
+            // Act
+            StateHolder.setCollection("newCollection", "value");
```

Comment:
`"value"` here should really be a collection, such as your mocked one. It's only a coincidence that a string happens to work instead.